src/dftbp/timedep/linrespcommon.F90

```diff
@@ -411,469 +406,464 @@ subroutine getSqrOcc(occNr, win, nmatup, nmat, getIA, tSpin, n_ij)
+    real(dp), intent(in) :: ovrXev(:,:,:)
+    
+    !> eigenvectors (nOrb, nOrb) [distributed]
+    real(dp), intent(in) :: grndEigVecs(:,:,:)
 
-    !> Onsite matrix elements for shells (elements between s orbitals on the same shell are ignored)
-    real(dp), intent(in), allocatable :: onsMEs(:,:,:,:)
+    !> DFTB gamma matrix (nAtm, nAtom)
+    real(dp), intent(in) :: frGamma(:,:)
 
-    !> Data type for atomic orbital information
-    type(TOrbitals), intent(in) :: orb
+    !> long-range Gamma if in use
+    real(dp), allocatable, optional, intent(in) :: lrGamma(:,:)
 
-    !> Whether (A+B)v or (Omega)v should be returned
+    !> whether (A+B)v or (Omega)v should be returned
     logical, intent(in) :: tAplusB
 
-    !> Machinery for transition charges between single particle levels
-    type(TTransCharges), intent(in) :: transChrg
-
-    !> Vector to multiply with size(nmat)
+    !> vector to multiply with size(nMat)
     real(dp), intent(in) :: vin(:)
 
-    !> Vector containing the result on exit size(nmat)
+    !> vector containing the result on exit size(nMat)
     real(dp), intent(out) :: vout(:)
 
-    !> Is calculation range-separated?
-    logical, intent(in) :: tHybridXc
-
-    !> Long-range Gamma if in use
-    real(dp), allocatable, optional, intent(in) :: lrGamma(:,:)
-
-    integer :: nmat, natom, norb, ia, nxvv_a, nSpin
+    integer :: nMat, nOrb, ia, nxvv_a
     integer :: ii, aa, ss, jj, bb, ias, jbs, abs, ijs, jas, ibs
+    integer :: nLoc, myia, myab, myja, nLocAB, iGlobalAB, fGlobalAB
+    integer, allocatable ::  getABasym(:,:)
 
-    ! Somewhat ugly, but fast small arrays on stack:
-    real(dp) :: otmp(size(gamma, dim=1)), gtmp(size(gamma, dim=1))
-    ! qij Working array (used for excitation charges). (nAtom)
-    real(dp) :: qij(size(gamma, dim=1))
+    ! somewhat ugly, but fast small arrays on stack:
+    real(dp) :: otmp(size(frGamma, dim=1)), gtmp(size(frGamma, dim=1))
+    real(dp) :: qij(size(frGamma, dim=1)) ! qij Working array (used for excitation charges). (nAtom)
     real(dp) :: vout_ons(size(vin))
-    real(dp) :: vTmp(size(vin))
-    real(dp), dimension(2) :: spinFactor = [1.0_dp, -1.0_dp]
-    ! For later use to change HFX contribution
+    real(dp), allocatable :: vLoc(:), vGlb(:), vGlb2(:)
+    real(dp), dimension(2) :: spinFactor = (/ 1.0_dp, -1.0_dp /)
+    ! for later use to change HFX contribution
     real(dp), allocatable :: qv(:,:)
 
     @:ASSERT(size(vin) == size(vout))
 
-    ! Dimension guessed from input vector
-    nmat = size(vin)
-    @:ASSERT(nmat <= nxov_rd)
-    natom = size(gamma, dim=1)
-    norb = size(ovrXev, dim=1)
-    @:ASSERT(norb == nocc_ud(1) + nvir_ud(1))
+    !! dimension guessed from input vector
+    nMat = size(vin)
+    @:ASSERT(nMat <= rpa%nxov_rd)
+    nOrb = orb%nOrb
 
+    !! Local chunk of RPA vectors have this size under MPI
+    nLoc = fGlobal - iGlobal + 1
+    @:ASSERT(nMat == nLoc)
+    allocate(vLoc, mold=vin)
+    vLoc = 0.0_dp
```

Comment:
```suggestion
    vLoc(:) = 0.0_dp
```
We always indicate array indices on the *left hand side* of assignments,, unless we want to trigger an explicit allocation/reallocation. (And we never indicate array indices on the *right hand side* unless necessary due to cutting out of sub-arrays)